TemporaryProjects.14.0/ProjectsLocationContext.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using EnvDTE;
+using Microsoft.VisualStudio.Shell;
+
+namespace TemporaryProjects
+{
+    public class ProjectsLocationEvent
+    {
+        readonly DTE dte;
+        readonly CommandEvents commandEvents;
+        IDisposable context;
+
+        public ProjectsLocationEvent(DTE dte, Guid guid, int id)
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+            this.dte = dte;
+            commandEvents = dte.Events.CommandEvents[guid.ToString("B"), id];
+            commandEvents.BeforeExecute += BeforeExecute;
+            commandEvents.AfterExecute += AfterExecute;
+        }
+
+        void BeforeExecute(string Guid, int ID, object CustomIn, object CustomOut, ref bool CancelDefault)
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+            context = new TempProjectLocationContext(dte);
```

Comment:
Hmm, I'm not completely sure whether this is the right thing to do. When the user has opted to save the project, it might be because they decided it should not be temporary after all. But on the other hand, they are forced to do this to overcome the nuget package limitation.

Reply:
It seems that this event isn't triggered when I click Save from the dialog shown after clicking Close Solution.